cli/explain/compactgraph/compactgraph.go

```diff
@@ -262,14 +262,8 @@ func (cg *CompactGraph) visitSuccessors(node any, visitor func(input any)) {
 			}
 		}
 	case *InputSet:
-		for _, file := range n.Files {
-			visitor(file)
-		}
-		for _, dir := range n.Directories {
-			visitor(dir)
-		}
-		for _, inputSet := range n.TransitiveSets {
-			visitor(inputSet)
+		for _, input := range n.Inputs {
+			visitor(input)
```

Comment:
Do you need an if/else here to support the old exec logs before 7.4?

If we don't intend to support the older version, we gonna need a warning message asking users to upgrade.

Reply:
`n` is a custom type here, not one of the protos, and `n.Inputs` contains the new and old proto fields merged together, so we really are supporting both versions.

Reply:
I see, so we pre-flatten the InputSet to reduce the nested loops inside the switch later. Nice.